src/homework1.ts

```diff
@@ -0,0 +1,139 @@
+class School {
+    directions: string[] = [];
+  
+    addDirection(direction: string): void {
+      this.directions.push(direction);
+    }
+  }
+  
+  class Direction {
+    levels: string[] = [];
+  
+    _firstName: string;
+  
+    get firstName(): string {
+      return this._firstName;
+    }
+  
+    constructor(name: string) {
+      this._firstName = name;
+    }
+  
+    addLevel(level: string) {
+      this.levels.push(level);
+    }
+  }
+  
+  class Level {
+    groups: string[] = [];
+  
+    _name!: string;
+    _program!: string;
+  
+    constructor(name: string, program: string) {
+      this._name = name;
+      this._program = program;
+    }
+  
+    get name(): string {
+      return this._name;
+    }
+  
+    get program() {
+      return this._program;
+    }
+  
+    addGroup(group: string): void {
+      this.groups.push(group);
+    }
+  }
+  
+  class Group {
+    _students: { name: string; getPerformanceRating: () => number; }[] = [];
+  
+    get students() {
```

Comment:
Please do not forget to add explicit type annotation for the return function value.